sdk/analytics/src/main/java/com/klaviyo/analytics/UserInfo.kt

```diff
@@ -22,14 +20,10 @@ import java.util.UUID
 internal object UserInfo {
 
     init {
-        Registry.get<ApiClient>().onApiRequest {
-            // Update state when a push token is successfully saved to backend
-            // Make sure it's still the same token we're tracking in state
-            if (it is PushTokenApiRequest &&
-                it.status == Complete &&
-                it.token == pushToken
-            ) {
-                setPushToken(pushToken, it.notificationPermission, it.backgroundData)
+        Registry.get<ApiClient>().onApiRequest { request ->
+            // If push token request totally fails, we must remove it from state
```

Comment:
Is this some mechanism to enforce us to send it again in the future since next time we call for push token it'll always be a value thats not empty?

Reply:
Exactly, this is the failure mode of the "optimistic update" approach. If a request fails to get to the backend, clearing pushState will mean a "duplicate" request could be enqueued again